### ISSUE TITLE
Stop Vendoring CUB and moderngpu

### ISSUE THREAD

**Issue by insertinterestingnamehere** [IntelligentSoftwareSystems/Galois#309]
CUB and moderngpu are both relatively standard GPU libraries. Does it make sense to continue vendoring them instead of requiring them as dependencies that are picked up through the usual CMake machinery?

**Comment by vishweshjatala** [comment]
I am okay with cub to specify it as a dependency requirement and specify the version to download. But, for moderngpu, we had to clone their master branch for latest fixes. So, this does not have any specific release. However, in my opinion, I prefer to continue to use the way we are using. @roshandathathri and @chenxuhao  please comment.

**Comment by roshandathathri** [comment]
No, I would prefer it the way it is. Moving to a new version of cub or moderngpu would require code changes as neither are guaranteed to maintain their API compatibility. Using it as a submodule gives us more control.
